### ISSUE TITLE
MIT personal websites are redirected to an incorrect URL 

### ISSUE THREAD

**Issue by cpitclaudel** [EFForg/https-everywhere#14397]
Type: ruleset issue
Domain: mit.edu

The ruleset in MIT.xml says this:

```xml
<rule from="^http://(www\.)?mit\.edu/" to="https://web.mit.edu/" />
```

This translates e.g. http://www.mit.edu/~cpitcla/test (which works) into https://web.mit.edu/~cpitcla/test (404).  Working HTTPS URLs for that page include https://www.mit.edu/~cpitcla/test (no need to rewrite the `www` part into `web`) and https://web.mit.edu/cpitcla/www/test (no `~`, extra `www`).

This is documented here: http://kb.mit.edu/confluence/pages/viewpage.action?pageId=3907090 .

It looks like the issue was introduced by https://github.com/EFForg/https-everywhere/pull/14169 or https://github.com/EFForg/https-everywhere/pull/12002 maybe?

```diff
+	<rule from="^http://(www\.)?mit\.edu/" to="https://web.mit.edu/" />
```

**Comment by jeremyn** [comment]
Yes, the `rule` in `MIT.xml` [here](https://github.com/EFForg/https-everywhere/blob/3c75f108300ca531d87395d3edbdb88c97631e26/src/chrome/content/rules/MIT.xml#L270) that you mentioned will rewrite http://www.mit.edu/~cpitcla/test to https://web.mit.edu/~cpitcla/test . That is a bad rewrite and should be fixed.

**Comment by Bisaloo** [comment]
I get different content for http://www.mit.edu/~cpitcla/test and https://www.mit.edu/~cpitcla/test

**Comment by cpitclaudel** [comment]
@Bisaloo Thanks for testing; looks like the https version is only available with MIT client certificates. I can confirm using curl instead of my browser.

**Comment by J0WI** [comment]
Thanks for your report. I just published a fix (#14417) which hopefully gets merged before the new release is made (#14336).
